### PR TITLE
Consolidate create/switch/update context with --connect

### PR DESCRIPTION
**Pull request by johannesfrey**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster ...

**What else do we need to know?**

### CODE REVIEW

**Review by FabianKramm — COMMENTED** [review]
@johannesfrey looks good to me, but I'm a little scared to remove the update-current flag, since it is referenced a lot. Can we basically just remove the functionality of it, but still have it as a hidden flag that is deprecated?

**Review by FabianKramm — APPROVED** [review]

File: pkg/cli/create_platform.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ func CreatePlatform(ctx context.Context, options *CreateOptions, globalFlags *fl
 	// check if we should connect to the vcluster or print the kubeconfig
 	if options.Connect || options.Print {
 		return ConnectPlatform(ctx, &ConnectOptions{
-			UpdateCurrent:         options.UpdateCurrent,
+			UpdateCurrent:         true,
```
Comment (rohantmp) [inline pkg/cli/create_platform.go:167]: I think ConnectPlatform and UpdateKubeconfig above duplicate some of the connection functionality

**Review by rohantmp — CHANGES_REQUESTED** [review]

File: pkg/cli/create_platform.go (hunk 1 above)
Comment (rohantmp) [reply pkg/cli/create_platform.go:167]: The `ConnectPlatform()` function ends with a `writkubeconfig()` function

File: pkg/cli/create_platform.go [hunk 2]
```diff
@@ -145,9 +145,9 @@ func CreatePlatform(ctx context.Context, options *CreateOptions, globalFlags *fl
 	}
 	log.Donef("Successfully created the virtual cluster %s in project %s", virtualClusterName, options.Project)
 
-	if options.CreateContext {
+	if options.Connect {
```
Comment (rohantmp) [inline pkg/cli/create_platform.go:148]: I wonder if this block could just be removed

File: pkg/cli/create_platform.go (hunk 1 above)
Comment (johannesfrey) [reply pkg/cli/create_platform.go:167]: Good point. I checked the code and it looks like `ConnectPlatform` does call `platform.CreateVirtualClusterInstanceOptions` internally, so essentially doing the same as the block L148-L162 you mentioned above. I went ahead and remove this block and checked `create` in the different combinations (`--connect=true`, `--connect=false`, `--connect=false --print=true` once without a platform and once with being logged into the platform). Looks good so far.

File: pkg/cli/create_platform.go (hunk 2 above)
Comment (johannesfrey) [reply pkg/cli/create_platform.go]: I removed it. See the comment above.

**Review by rohantmp — APPROVED** [review]
LGTM :)

**Review by FabianKramm — APPROVED** [review]

**Review by rohantmp — APPROVED** [review]

**Pull request merged**